PowerPointLabs/PowerPointLabs/ThisAddIn.cs

```diff
@@ -834,33 +880,16 @@ private void ThisAddInStartup(object sender, EventArgs e)
 
         private void ThisAddInApplicationOnWindowDeactivate(PowerPoint.Presentation pres, PowerPoint.DocumentWindow wn)
         {
-            Trace.TraceInformation(pres.Name + " terminating...");
-            Trace.TraceInformation(string.Format("Is Closing = {0}, Count = {1}", _isClosing,
-                Application.Presentations.Count));
-
+            Trace.TraceInformation(pres.Name + " (Presentation) and " + wn.Caption + " (Window) deactivated.");
             _deactivatedPresFullName = pres.FullName;
-
-            // in this case, we are closing the last client presentation,
-            // therefore we can close the shape gallery
-            if (_isClosing &&

```

Comment:
I noticed that the presentations (this whole code is not run) are now not closed OnWindowDeactivate, is it confirmed okay?

Reply:
Yup, this is ok because `OnWindowDeactivate()` is of lower priority than `PresentationClose()` (This priority is defined in `ThisAddInStartup()`). Upon clicking 'X' on the main window, `PresentationClose()` is called and all background ppts are closed here. By the time code execution reaches `OnWindowDeactivate()`, all background ppt should be closed already.